pkg/kv/kvserver/mvcc_gc_queue.go

```diff
@@ -421,9 +422,26 @@ func makeMVCCGCQueueScoreImpl(
 		r.FinalScore++
 	}
 
+	// If range was deleted, lower score threshold to allow faster cleanup as
+	// most of the garbage is concentrated at the ttl threshold.
+	if !r.ShouldQueue && suspectedFullRangeDeletion(ms) {
+		r.ShouldQueue = canAdvanceGCThreshold && r.FuzzFactor*valScore > mvccGCDropRangeKeyScoreThreshold
+	}
+
 	return r
 }
 
+// suspectedFullRangeDeletion checks for ranges where there's no live data and
+// range tombstones are present. This is an indication that range is likely
+// removed by bulk operations and its garbage collection should be done faster
+// than if it has to wait for double ttl.
+func suspectedFullRangeDeletion(ms enginepb.MVCCStats) bool {
+	if ms.LiveCount > 0 || ms.IntentCount > 0 {
+		return false
+	}
+	return ms.RangeKeyCount > 0
```

Comment:
nit: I believe intents are already accounted for in `LiveCount`, so this simplifies to `return ms.LiveCount == 0 && ms.RangeKeyCount > 0`.

We're not checking estimated stats here, but that's probably fine.

Reply:
Hmm, is it correct with intent counts? When we delete in transaction we seem to decrease live count, but increase intent count when it is created if we covered previous value.

Reply:
Ah, you're right, I didn't take into account tombstone intents.